tests/framework/microvm.py

```diff
@@ -505,6 +505,7 @@ def spawn(
         log_show_level=False,
         log_show_origin=False,
         metrics_path="fc.ndjson",
+        force_new_pid_ns=False,
```

Comment:
There is no reason to have this here since it is hardcoded later in the test right?

Reply:
This is juts to have less code changes for tests like [serial](https://github.com/sudanl0/firecracker/blob/3a7d32ca42b9d6fe668c5874ea018ca7e4d964a3/tests/integration_tests/functional/test_serial_io.py#L142) where the daemonize is set to false just before we spawn the microvm.

We cannot change the behaviour of Jailer after the spawn() and we need to test the condition "daemonize=False but new_pid_ns=True" so this flag was needed.

Reply:
Is that a combination that works? Could we just remove the check in line 547 then?

Reply:
yes the combination only works if we have line 547.
We have all these places where test explicitly sets the daemonize to False before spawn() https://github.com/search?q=repo%3Afirecracker-microvm%2Ffirecracker+%22jailer.daemonize+%3D+False%22&type=code 
If we remove the force_new_pid_ns flag and line 547 then we'll have to update all these tests to explicitly set new_pid_ns as False and I think it would only make the code look more ugly and increase chances of human error.

Using the force_new_pid_ns flag we keep the behaviour of all the other tests same and make sure that the new test `test_firecracker_kill_by_pid` also works correctly.

Reply:
I disagree there. Tests should be explicit about what they want than depend on some (arbitrary) policy done within the Microvm class. If the combination works, there should be a clean way to ask Microvm to do that. We may need to change surrounding code and tests to match the new behavior, but that is more maintainable than adding flags to force behavior.

Reply:
I think there is some misunderstanding here, the tests **already** depended on some policy that was done in the Microvm class. The existing assumption in the tests is that if daemonize is False then new_pid_ns should be False as well. If we want to make this explicit then we'll have to remove this assumption and update all the tests that depended on it. Updating all the other tests is what I was trying to avoid since it looked like an unrelated change but I have added another commit to make that explicit.
I didn't use jailer_kwargs in the new commit because, some tests used the fixture uvm_plain, some used uvm_plain_initrd etc. so using jailer_kwargs means that we'll need to modify all these fixtures and I would like to avoid more code change at this point since it falls into cleanup category and out of scope for the current task.